dokka-integration-tests/gradle/src/test/kotlin/AndroidProjectIT.kt

```diff
@@ -0,0 +1,163 @@
+                "--rerun-tasks",
+            )
+            .build {
+                withClue("expect project builds successfully") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(SUCCESS)
+                }
+
+                withClue("expect all dokka workers are successful") {
+                    project
+                        .findFiles { it.name == "dokka-worker.log" }
+                        .shouldForAll { dokkaWorkerLog ->
+                            dokkaWorkerLog.shouldBeAFile()
+                            dokkaWorkerLog.readText().shouldNotContainAnyOf(
+                                "[ERROR]",
+                                "[WARN]",
+                            )
+                        }
+                }
+
+                withClue("expect configurations are not resolved during configuration time") {
+                    output shouldNotContain Regex("""Configuration '.*' was resolved during configuration time""")
+                    output shouldNotContain "https://github.com/gradle/gradle/issues/2298"
+                }
+            }
+
+        withClue("expect the same HTML is generated") {
+            val expectedHtml = project.projectDir.resolve("expectedData/html")
+            val actualHtmlDir = project.projectDir.resolve("build/dokka/html")
+
+            withClue(
+                """
+                expectedHtml: ${expectedHtml.toUri()}
+                actualHtmlDir: ${actualHtmlDir.toUri()}
+                """.trimIndent()
+            ) {
+                val expectedFileTree = expectedHtml.toTreeString()
+                val actualFileTree = actualHtmlDir.toTreeString()
+                withClue((actualFileTree to expectedFileTree).sideBySide()) {
+                    expectedFileTree shouldBe actualFileTree
+
+                    actualHtmlDir shouldBeADirectoryWithSameContentAs expectedHtml
+                }
+            }
+        }
+    }
+
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android")
+    fun `Dokka tasks should be build cacheable`(
+        project: DokkaGradleProjectRunner,
+    ) {
+        project.runner
+            .addArguments(
+                ":dokkaGenerate",
+                "--build-cache",
+            )
+            .build {
+                withClue("expect dokkaGenerate runs successfully") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(UP_TO_DATE, SUCCESS)
+                }
+            }
+
+        project.runner
+            .addArguments(
+                ":dokkaGenerate",
+                "--build-cache",
+            )
+            .build {
+                withClue("expect dokkaGenerate tasks are loaded from cache") {
+                    shouldHaveTask(":dokkaGenerate").shouldHaveOutcome(UP_TO_DATE)
+                }
+            }
+    }
+
+    @DokkaGradlePluginTest(sourceProjectName = "it-android")
+    fun `expect Dokka is compatible with Gradle Configuration Cache`(
```

Comment:
also it looks like AndroidComposeIT and AndroidProjectIT have shared code. Is it ok for you to keep those duplicated? i.e. you don't see lots of maintenance with this code in the future.